### ISSUE TITLE
Bash argument completion for tensorboard

### ISSUE THREAD

**Issue by Garoe** [tensorflow/tensorflow#6479]
It be nice to have autocompletion for the tensorboard arguments.
I experimented a bit https://github.com/Garoe/tensorflow/commit/3c217767cd47560c776232207a05b08b6f2cb166 with the [argcomplete](https://argcomplete.readthedocs.io/en/latest/), however the completion is too sluggish to be useful.
The entry point for the completion is after the flags are defined and the slugginesh is due to too much stuff being done before that.
Another option would be to use [native completion](https://github.com/scop/bash-completion), but that would require to duplicate the flag definitions for tensorboard.

**Comment by aselle** [comment]
@dnamane, please consider this feature request. Thanks!

**Comment by aselle** [comment]
er @danmane

**Comment by Garoe** [comment]
I've put together a simple file using the native option https://github.com/Garoe/tensorflow/commit/887678ef4ed1da04a13c272e5d5bb8d5fc2ac536
For installation without root access, the file needs to be copied to  `~/.bash_completion.d/tensorboard` and the following line `. ~/.bash_completion.d/tensorboard` has to be added to a `~/.bash_completion` text file.
I'd be happy to add the steps in the install process and create a proper pull request, but I'd really appreciate some pointers into what files should be modified as I'm not familiar with Bazel.

**Comment by dandelionmane** [comment]
Hmm, I think the installation should happen in the pip package setup, rather than in Bazel. 
I suspect either [this file](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/pip_package/setup.py) or something else in the pip_package directory.

@vrv care to comment?

**Comment by vrv** [comment]
Honestly, I'd prefer this live as a nice external repository we point to, which we can link to somewhere.

**Comment by dandelionmane** [comment]
If it's in an external repository, would we depend on it and install it by default, or would this be only for people who really really care about bash completion and go through the work of installing it themselves?

**Comment by zaccharieramzi** [comment]
Hey!
So what is the status of this feature request?

**Comment by Garoe** [comment]
I've put the file in a external repository and it's updated for the parameters used in version 1.2
As I said before, I'd happy to write a pull request to include this by default or however they think it's best, so I'm still waiting on someone to answer @dandelionmane  question.

https://gist.github.com/Garoe/e6c8a3a0af6f487f2886c77f2a185738
